lib/logstash/outputs/influxdb.rb

```diff
@@ -126,30 +144,52 @@ def receive(event)
     # ]
     event_hash = {}
     event_hash['name'] = event.sprintf(@series)
-    sprintf_points = Hash[@data_points.map {|k,v| [event.sprintf(k), event.sprintf(v)]}]
-    if sprintf_points.has_key?('time')
-      @logger.error("Cannot override value of time without 'allow_override_time'. Using event timestamp") unless @allow_override_time
+
+    if !use_event_fields_for_data_points
+		logger.debug("#{use_event_fields_for_data_points} is false. Using preconfigured data points to build influxdb request")
+		sprintf_points = Hash[@data_points.map {|k,v| [event.sprintf(k), event.sprintf(v)]}]
+    return
     else
-      sprintf_points['time'] = to_epoch(event.timestamp)
+		logger.debug("#{use_event_fields_for_data_points} is true. Using fields from event to build influxdb request")
+    	if fields_to_skip.any?
+       		array_to_skip = fields_to_skip
+    	else
+        	array_to_skip = []
+    	end
+		
+	   logger.debug("Those fields going to be skipped from influxdb request #{array_to_skip.to_json} ")
+       sprintf_points = Hash[event] 
+       sprintf_points.delete_if{|k,v| array_to_skip.include?(k)}
+	   logger.debug("Those fields going to be used for influxdb request #{sprintf_points.to_json} ")
     end
-    @coerce_values.each do |column, value_type|
-      if sprintf_points.has_key?(column)
-        begin
-          case value_type
-          when "integer"
-            @logger.debug("Converting column #{column} to type #{value_type}: Current value: #{sprintf_points[column]}")
-            sprintf_points[column] = sprintf_points[column].to_i
-          when "float"
-            @logger.debug("Converting column #{column} to type #{value_type}: Current value: #{sprintf_points[column]}")
-            sprintf_points[column] = sprintf_points[column].to_f
-          else
-            @logger.error("Don't know how to convert to #{value_type}")
-          end
-        rescue => e
-          @logger.error("Unhandled exception", :error => e.message)
-        end
-      end
+     
+    if sprintf_points.has_key?('time')
+      @logger.error("Cannot override value of time without 'allow_time_override'. Using event timestamp") unless @allow_time_override
+    else
+     # sprintf_points['time'] = to_epoch(event.timestamp)
+      sprintf_points['time'] = event.timestamp.to_i
```

Comment:
You corrected the time_precision to use "ms" instead of "m" however since you're using `to_i` here, you'll never get millisecond precision. [See my fix for this](https://github.com/logstash-plugins/logstash-output-influxdb/pull/23/files#diff-a6c25cf1ec24b5b0f50ad123690852f7R232)